### PR TITLE
Update get_obs_surface to allow inlet to be skipped when not needed

### PR DESCRIPTION
**Pull request by rt17603**
Based on Issue #119, this PR is to allow the ranking to be skipped when not needed. This means `get_obs_surface` can be called as follows:

`get_obs_surface(site="tac", species="ch4")`

and as long as there are not multiple inlets, the single data source can be successfully retrieved. An explicit input for `skip_ranking` has also been added e.g.

`get_obs_surface(site="tac", species="ch4", skip_ranking=True)`

### CODE REVIEW

**Comment by rt17603** [comment]
Need to fix test outcomes that have been changed based on this PR especially:
=========================== short test summary info ============================
FAILED tests/dataobjects/test_searchresults.py::test_retrieve_complex_ranked
======= 1 failed, 145 passed, 7 skipped, 49 warnings in 79.87s (0:01:19) =======

**Comment by rt17603** [comment]
Now fixed tests - but this is fixed in relation to the new logic in this PR. So overall we need to check we're happy with this logic.

File: tests/retrieve/test_observations.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+def test_get_obs_surface_one_inlet():
+    '''
+    Test we can access site and species data without needing to specify the inlet
+    when this is not needed.
+    '''
+    # Rely on data already loaded from conftest.py and in local object store
+    data = get_obs_surface(site="tac", species="ch4")
+
+    assert isinstance(data, ObsData)
```
Comment (gareth-j) [inline tests/retrieve/test_observations.py:12]: I've added in a few extra asserts to make sure we get the correct data back here.

**Review by gareth-j — APPROVED** [review]
Looks good, I've updated the test.

**Pull request merged**